### DISCUSSION TITLE
doubt in DataListLoader

### DISCUSSION THREAD

**Discussion by steffyalbert** [pyg-team/pytorch_geometric#2999] [Q&A]
I am using a custom dataset of fluid simulation and I am using the dataloader as follows:

![image](https://user-images.githubusercontent.com/8213883/130158266-cc18206f-527e-4814-a4ee-79238bb44af4.png)

Using a custom collate function as above, but I am getting the error as
![image](https://user-images.githubusercontent.com/8213883/130157869-1bdc8b9a-0e83-4e95-9882-11ae0a60f425.png)

How can resolve this TypeError: can only concatenate list (not "Data") to list?

**Comment by rusty1s** [comment]
What shall `l = sum(items, []) be doing? In order for the call to succeed, it needs to be passed a list of lists. Otherwise:
```python
def collate(items):
    return Batch.from_data_list(items)
```
should work just fine (and is equivalent to loading via `DataLoader`).
